### PR TITLE
Gui changes

### PR DESCRIPTION
**Pull request by marauder2k7**
-Check for profile in GuiControl
-Changed onMouse for onTouch
-Profiles for more classes

### CODE REVIEW

**Comment by marauder2k7** [comment]
added debug 64bit changes to the pull request, only windows atm and only debug

File: engine/source/platform/types.gcc.h [hunk 1]
```diff
@@ -55,7 +55,12 @@ typedef unsigned long long  U64;
...
 //--------------------------------------
 // Identify the Operating System
-#if defined(__WIN32__) || defined(_WIN32)
+#if defined(_WIN64)
+#  define TORQUE_OS_STRING "Win64"
+#  define TORQUE_OS_WIN
+#  define TORQUE_OS_WIN64
+#  include "platform/types.win32.h"
+#elif defined(__WIN32__) || defined(_WIN32)
 #  define TORQUE_OS_STRING "Win32"
```
Comment (dottools) [inline engine/source/platform/types.gcc.h:64]: No `TORQUE_OS_WIN` for Windows 32bit build using GCC?

**Comment by dottools** [comment]
Downside to renaming onMouse to onTouch is that there are still several variables using onMouse. And that onTouchEnter and onTouchLeave don't make sense for touch events considering those are mouse only events as they pertain to the mouse cursor moving in and out of the container firing off those events.

File: engine/source/platform/types.gcc.h (hunk 1 above)
Comment (marauder2k7) [reply engine/source/platform/types.gcc.h:64]: TORQUE_OS_WIN is defined for 32bit in types.visualc 

Here for 64BIT TORQUE_OS_WIN probably shouldn't be defined or add it under WIN32 either way it does get set.

**Pull request merged**